vault/logical_passthrough.go

```diff
@@ -49,6 +48,7 @@ func LeaseSwitchedPassthroughBackend(conf *logical.BackendConfig, leases bool) (
 				HelpDescription: strings.TrimSpace(passthroughHelpDescription),
 			},
 		},
+		BackendType: backendType,
```

Comment:
I am using `logical.BackendType` to determine whether the passthrough backend needs to generate leases or not. This eliminates the need for the `GenerateLeases()` method and consequently the need to perform type assertion on `handleRequest(...)` in `request_handling.go`. More importantly, this allows `handleRequest(...)` to perform the same check for a passthrough backend plugin.